libnavigation-core/src/main/java/com/mapbox/navigation/core/arrival/AutoArrivalController.kt

```diff
@@ -0,0 +1,41 @@
+package com.mapbox.navigation.core.arrival
+
+import android.os.SystemClock
+import com.mapbox.api.directions.v5.models.RouteLeg
+import com.mapbox.navigation.base.trip.model.RouteLegProgress
+import java.util.concurrent.TimeUnit
+
+/**
+ * The default controller for arrival. This will move onto the next leg automatically
+ * if there is one.
+ */
+class AutoArrivalController : ArrivalController {
+
+    private var routeLegCompletedTime: Long? = null
+    private var currentRouteLeg: RouteLeg? = null
+
+    /**
+     * Default arrival options.
+     */
+    override fun arrivalOptions(): ArrivalOptions = ArrivalOptions.Builder().build()
+
+    /**
+     * By default this will move onto the next step.
```

Comment:
Could we add a note around `arrivalInSeconds` (e.g. `when arrivalInSeconds have passed`)? If I remember correctly, `arrivalInSeconds` defaults to `5` seconds.